### ISSUE TITLE
SVG: Unable to specify an alt text for screen readers

### ISSUE THREAD

**Issue by davidmurray** [Kozea/WeasyPrint#2816]
Hello,

I have an SVG image embedded in my HTML file. I want to provide an alt text so that screen readers will provide a description of the image when reading the PDF out loud.

I have provided the <title> tag in my SVG structure:
```xml
<svg xmlns:xlink="http://www.w3.org/1999/xlink" width="288pt" height="288pt" viewBox="0 0 288 288" xmlns="http://www.w3.org/2000/svg" version="1.1"><title>Tracé du parcours de la ligne 21.</title>...
```
and as you can see, it's being picked up by the browser:

<img width="864" height="346" alt="Image" src="https://github.com/user-attachments/assets/53f9f465-2162-426d-ac69-a77a8f69cbe0" />

However, in the PDF file created by Weasyprint, I don't see any title associated with the SVG file. Nothing is being read by the screen reader.

<img width="1321" height="553" alt="Image" src="https://github.com/user-attachments/assets/f69c6a29-4b5d-468e-b3fc-e1af19d1968a" />

**What is the best way to provide an alt text for svg images?**

Here are the HTML and PDF files:
[SO21-20260106_21001231 (12).html](https://github.com/user-attachments/files/29473972/SO21-20260106_21001231.12.html)
[SO21-20260106_21001231 (26).pdf](https://github.com/user-attachments/files/29473973/SO21-20260106_21001231.26.pdf)

Thanks

**Comment by liZe** [comment]
Thanks for the report, that’s another accessibility issue we should fix!

**Comment by liZe** [comment]
If anyone is interested in solving this issue while learning WeasyPrint internals (ie without AI), I can help!

**Comment by davidmurray** [comment]
Hi @liZe, I could give this a shot if you gave me some tips on where to look in the codebase.
Do you think issue https://github.com/Kozea/WeasyPrint/issues/2550 would be somewhat easy to implement as well?
Thanks

**Comment by liZe** [comment]
Hi @davidmurray,

For this issue, you can:

- set the 'Figure' PDF tag in ['_get_pdf_tag`](https://github.com/Kozea/WeasyPrint/blob/bfa1bb8c6ff8005116df4e1c8f3e9170ce0d02a9/weasyprint/pdf/tags.py#L80),
- set the ['Alt' property](https://github.com/Kozea/WeasyPrint/blob/bfa1bb8c6ff8005116df4e1c8f3e9170ce0d02a9/weasyprint/pdf/tags.py#L199) the the element. You can [find](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.find) the 'title' tag in `box.element`.

Note that tags in SVG have a namespace. For example, the `svg` tag name is `"{http://www.w3.org/2000/svg}svg"`, not just `"svg"`.

#2550 shouldn’t be really complex, but it requires a lot of work to do things well: we’ll have to read the [ARIA spec](https://w3c.github.io/aria/) and the PDF spec to list all the different ARIA roles and map them to corresponding PDF tags.
